app/api/v1/endpoints/category.py

```diff
@@ -0,0 +1,115 @@
+from fastapi import APIRouter, Depends, HTTPException, status
+from app.schemas.category import CategoryCreate, CategoryResponse, CategoryUpdate
+from app.schemas.base import ResponseBase
+from app.services.category import (
+    add_category,
+    select_categories,
+    select_category_by_id,
+    select_category_by_name,
+    update_category_by_id,
+    delete_category_by_id,
+)
+from sqlalchemy.ext.asyncio import AsyncSession
+from app.db.session import get_db
+from typing import List
+from uuid import UUID
+
+router = APIRouter()
+
+
+@router.post("/", response_model=ResponseBase[CategoryResponse])
+async def post_category(
+    category_data: CategoryCreate,
+    db: AsyncSession = Depends(get_db),
+) -> ResponseBase[CategoryResponse]:
+    try:
+        category = await add_category(db, category_data)
+        return ResponseBase(status_code=status.HTTP_201_CREATED, data=category)
+    except HTTPException as e:
+        return ResponseBase(status_code=e.status_code, error=e.detail)
+    except Exception as e:
+        return ResponseBase(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, error=str(e)
+        )
+
+
+@router.get("/", response_model=ResponseBase[List[CategoryResponse]])
+async def get_categories(
+    db: AsyncSession = Depends(get_db),
+) -> ResponseBase[List[CategoryResponse]]:
+    try:
+        categories = await select_categories(db)
+        return ResponseBase(status_code=status.HTTP_200_OK, data=categories)
+    except HTTPException as e:
+        return ResponseBase(status_code=e.status_code, error=e.detail)
+    except Exception as e:
+        return ResponseBase(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, error=str(e)
+        )
+
+
+@router.get("/{category_id}", response_model=ResponseBase[CategoryResponse])
+async def get_category_by_id(
+    category_id: UUID,
+    db: AsyncSession = Depends(get_db),
+) -> ResponseBase[CategoryResponse]:
+    try:
+        category = await select_category_by_id(db, category_id)
+        return ResponseBase(status_code=status.HTTP_200_OK, data=category)
+    except HTTPException as e:
+        return ResponseBase(status_code=e.status_code, error=e.detail)
+    except Exception as e:
+        return ResponseBase(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, error=str(e)
+        )
+
+
+@router.get("/name/{category_name}", response_model=ResponseBase[CategoryResponse])
```

Comment:
@Ryan-OH 
보통 명사 자원에는 복수형으로 해주는게 자연스러운거 같습니다. names 중에서도 특정이름 인거라고 봐서!
```suggestion
@router.get("/names/{category_name}", response_model=ResponseBase[CategoryResponse])
```

Reply:
오호 이부분도 명사자원은 복수형으로 전환하겠습니다